url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+         //y no hay que generar las capturas ni las intercepciones
+         if(numAttackers == 1)
+         {
+            //Si sólo está dando jaque una pieza se generan las capturas de la pieza atacante y los movimientos de intercepción
+            
+            checkSquareIndex = Long.numberOfTrailingZeros(bbKingAttackers);
+            /*** GENERACIÓN DE LAS CAPTURAS DE LA PIEZA ATACANTE ****/
+            
+            //Se recuperan las piezas que atacan a la pieza que ejerce el jaque
+            bbAttackers = bbAttackerToSquare[checkSquareIndex];
+            //Se quitan los atacantes del turno contrario y el propio rey
+            bbAttackers &= ~(bbOppositeOccupied | bbKing) ; 
+            
+            numAttackers = Long.bitCount(bbAttackers);
+            for(int i = 0; i < numAttackers; i++)
+            {
+               pieceSquareIndex = Long.numberOfTrailingZeros(bbAttackers);
+               
+               isPinned = false;
+               
+               bbAttacked = bbKingAttackers;
+               
+               //Se comprueba si hay conexión entre la casilla del rey y la casilla de la pieza
+               //Si no la hay se generan los movimientos de forma normal
+               bbConnection = bbInclusiveConnection[pieceSquareIndex][kingSquareIndex];
+               if(bbConnection != 0)
+               {
+                  //Si hay conexión, se comprueba si hay alguna pieza en las casillas
+                  //que unen la casilla del rey y la casilla de la pieza
+                  //Si sí que hay alguna pieza se generan los movimientos de la pieza de forma normal
+                  bbConnection = bbExclusiveConnection[pieceSquareIndex][kingSquareIndex];
+                  if((bbConnection & bbOccupation) == 0)
+                  {
+                     //Si no hay casillas se comprueba si hay alguna pieza que ataque a la pieza en la misma
+                     //linea que el rey y la pieza
+                     //Si no hay ninguna se generan los movimientos de la pieza de forma normal
+                     bbAttacker = bbFullConnection[pieceSquareIndex][kingSquareIndex] & 
+                           bbAttackerToSquare[pieceSquareIndex] & bbOppositeOccupied;
+                     if(bbAttacker != 0)
+                     {
+                        genericPiece = pieceInSquare[Long.numberOfTrailingZeros(bbAttacker)].genericPiece;
+                        if(genericPiece != GenericPiece.PAWN && genericPiece != GenericPiece.KING)
+                        {
+                           //Si hay alguna pieza atacando en esa linea quiere decir que la pieugggza
+                           //sólo se podrá mover a las casillas que conectan las casilla de la pieza
+                           //y la casilla del atacante.
+                           isPinned = true;
+                        }
+                     }
+                  }
+               }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1815:1840, 1905:1927, 2118:2143, 2249:2274.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.